packages/grid_client/scripts/applications/peertube.ts

```diff
@@ -0,0 +1,115 @@
+import { FilterOptions, GatewayNameModel, MachinesModel } from "../../src";
+import { config, getClient } from "../client_loader";
+import { log, pingNodes } from "../utils";
+
+async function deploy(client, vms, subdomain, gatewayNode) {
+  const resultVM = await client.machines.deploy(vms);
+  log("================= Deploying VM =================");
+  log(resultVM);
+  log("================= Deploying VM =================");
+
+  const vmPlanetary = (await client.machines.getObj(vms.name))[0].planetary;
+  //Name Gateway Model
+  const gw: GatewayNameModel = {
+    name: subdomain,
+    node_id: gatewayNode.nodeId,
+    tls_passthrough: false,
+    backends: ["http://[" + vmPlanetary + "]:9000"],
+  };
+
+  const resultGateway = await client.gateway.deploy_name(gw);
+  log("================= Deploying name gateway =================");
+  log(resultGateway);
+  log("================= Deploying name gateway =================");
+}
+
+async function getDeployment(client, vms, gw) {
+  const resultVM = await client.machines.getObj(vms.name);
+  const resultGateway = await client.gateway.getObj(gw);
+  log("================= Getting deployment information =================");
+  log(resultVM);
+  log(resultGateway);
+  log("https://" + resultGateway[0].domain);
+  log("================= Getting deployment information =================");
+}
+
+async function cancel(client, vms, gw) {
+  const resvm = await client.machines.delete(vms);
+  const resgw = await client.gateway.delete_name(gw);
+  log("================= Canceling the deployment =================");
+  log(resvm);
+  log(resgw);
+  log("================= Canceling the deployment =================");
+}
+
+async function main() {
+  const name = "newpeertube";
+  const grid3 = await getClient(`peertube/${name}`);
+  const subdomain = "pt" + grid3.twinId + name;
+
+  //VMNode Selection
+  const vmQueryOptions: FilterOptions = {
+    cru: 1,
+    mru: 2,
+    sru: 50,
+    availableFor: grid3.twinId,
+    farmId: 1,
+  };
+  //GatewayNode Selection
+  const gatewayQueryOptions: FilterOptions = {
+    gateway: true,
+    availableFor: grid3.twinId,
+  };
+  const gatewayNode = (await grid3.capacity.filterNodes(gatewayQueryOptions))[0];
+
+  const vms: MachinesModel = {
+    name,
+    network: {
+      name: "wedtest",
+      ip_range: "10.249.0.0/16",
+    },
+    machines: [
+      {
+        name: "peertube",
+        node_id: +(await pingNodes(grid3, await grid3.capacity.filterNodes(vmQueryOptions))),
```

Comment:
Instead of calling the function inside the object, what do you think if we stashed the value of it in a const and then used it inside the object? I think it'll enhance the code's readability.

if you agree with me, please change it in all of your scripts :)


```ts
const options = {...}
const nodes = await grid3.capacity.filterNodes(options)
const selectedNodeID = await pingNodes(nodes)

// then use it 

machines: [
  {
     node_id: selectedNodeID
  }
]
```